src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java

```diff
@@ -93,7 +98,14 @@ public JsonResult execute() {
             }
         } else {
             feedbackSessionAttributes = logic.getFeedbackSessionsForCourse(courseId);
-            if (entityType.equals(Const.EntityType.INSTRUCTOR)) {
+            if (entityType.equals(Const.EntityType.STUDENT) && !feedbackSessionAttributes.isEmpty()) {
+                StudentAttributes student = logic.getStudentForGoogleId(courseId, userInfo.getId());
+                assert student != null;
```

Comment:
Is this always true? I recall relevant discussion on this subject and the conclusion was this should not happen. Was overwriting `feedbackSessionAttributes` to empty list a viable solution that honours the contract of this API?

Reply:
Do you mean the `student != null` part? I think it would be null if `courseId` is invalid (or if the user is not a student, but we agreed to assume this is true). The check `!feedbackSessionAttributes.isEmpty()` ensures `courseId` is valid.

Reply:
Yup. I recall that too, where it is mentioned the student can be an admin for some weird reason. @samuelfangjw seems to have handled this just in case. I would think an assert here is appropriate then, whereby we are alerted should such a weird situation happens.

Reply:
Yes, just looking at this class in isolation there is a path where the `userId` may not be the students if the admin is the one requesting for information, although I have a feeling this is not an issue, since `instructors = Collections.singletonList(logic.getInstructorForGoogleId(courseId, userInfo.getId()));` exists and this uses `userInfo.getId()` as well. However, in this case it also makes sense to not hide information if user is an admin, so no harm making the change.